test: Add test to ensure unknown algorithms do not break verify_event

gnieto opened the pull request:
The event is properly signed, but key map contains a key with an unknown algorithm. Under this circumstances, the event should be allowed.

The test will fail until https://github.com/ruma/ruma/pull/1496 gets merged

<!-- Replace -->
----
Preview Removed
<!-- Replace -->

gnieto committed:
test: Add test to verify unknown algorithms do not break verify events

The event is properly signed, but key map contains a key with an
unknown algorithm. Under this circumstances, the event should be
allowed.

ShadowJonathan commented:
> Under these circumstances, the event should be allowed.

Why? I imagine that the best fail-safe method of dealing with an event from a potentially-malicious agent (with an unknown signature algorithm) is to simply drop it?

What does the spec say about this?

gnieto commented:
2nd point in this section of the spec explains that unknown algorithms should be filtered: https://spec.matrix.org/v1.6/appendices/#checking-for-a-signature

gnieto committed:
signatures: Add more test for latest changes in verify_event

Cover new cases which were not covered before:

- An event signed by multiple keys for an entity, should verify all
  those signatures
- An event signed by a single key with an unkown algorithm by the
  required entity, should fail

gnieto commented:
Superseded by: https://github.com/ruma/ruma/pull/1503
